SFS_general/delayline.m

```diff
@@ -1,149 +1,221 @@
+if ndims(sig)==3
+    [M, C, samples] = size(sig);
+    channels = M * C;
+    % Reshape [M C N] => [N C*M], this will be redone at the end of the function
+    sig = reshape(sig,[channels,samples])';
+    reshaped = true;
+else
+    % Assume standard format [N C]
+    [samples, channels] = size(sig);
+    reshaped = false;
+end
+
+% If only single valued time delay and weight is given, create vectors
+if channels>1 && length(dt)==1, dt=repmat(dt,[1 channels]); end
+if channels>1 && length(weight)==1, weight=repmat(weight,[1 channels]); end
+
+rfactor = 1.0;  % ratio of signal length and delayline length
+switch fracdelay.pre.method
+    case 'resample'
+        % === Resample ===================================================
+        % resample factor (1/stepsize of fractional delays)
+        rfactor = fracdelay.pre.resample.factor;
+        switch fracdelay.pre.resample.method
+            case 'matlab'
+                sig = resample(sig,rfactor,1);
+            case 'pm'
+                % === Parks-McClellan linear phase FIR filter ===
+                a = [1 1 0 0];
+                f = [0.0 0.9/rfactor 1/rfactor 1.0];
+                b = firpm(fracdelay.pre.resample.order,f,a);
+
+                sig = reshape(sig,1,channels*samples);
+                sig = [sig; zeros(rfactor-1,channels*samples)];
+                sig = reshape(sig,rfactor*samples,channels);
+                
+                sig = filter(b,1,sig,[],1);
+            otherwise
+                disp('Delayline: Unknown resample method');
+        end
+    case 'farrow'
+        % === Farrow-Structure ===========================================
+        % Based on the assumption, that each coefficient h(n) of the fractional
+        % delay filter can be expressed as a polynomial in d (frac. delay), i.e.
+        %            _
+        %           \  NPol
+        % h_d(n) ~=  >      c_m(n) d^m
+        %           /_ m=0
+        %
+        % For some Filter design methods, e.g. Lagrange Interpolators, this
+        % perfectly possible. For other, a uniform grid of test delays d_q is
+        % used to fit the polynomials to the desired coefficient(n) find a set
+        % polynomial which approximates each coefficient of the desired filter.
+        % This structure allows to perform the convolution independently from 
+        % the delay and reuse the results of the filter for different delays.
+        %                          _
+        %                          \  NPol
+        % y(n) = h_d(n) * x(n) ~=   >      ( c_m(n)*x(n) ) d^m
+        %                          /_ m=0
+        %
+        % The above representation shows that the convolution of the input 
+        % signal x can be performed by first convolving c_m and x and 
+        % incorporating the delay d afterwards.
+        
+        % number of parallel filters, i.e. order of polynomial + 1
+        % Nfilter = fracdelay.pre.farrow.Npol+1;
+        
+        to_be_implemented(mfilename);
```

Comment:
OK, perfect.
Then we postpone this after this pull request and I will merge this section as it is currently.